NOnion/Constants.fs

```diff
@@ -18,3 +18,61 @@ module Constants =
     (* Amount of bytes needed for generating keys and digest during KDF = 2 * KeyLength + 3 * HashLength *)
     [<Literal>]
     let KdfLength = 92
+
+    let SupportedProtocolVersion: array<uint16> = [| 3us |]
+
+    (*
+     *  Existing Tor implementations choose their CircID values at random from
+     *  among the available unused values.  To avoid distinguishability, new
+     *  implementations should do the same. Implementations MAY give up and stop
+     *  attempting to build new circuits on a channel, if a certain number of
+     *  randomly chosen CircID values are all in use (today's Tor stops after 64).
+     *)
+
+    [<Literal>]
+    let MaxCircuitIdGenerationRetry = 64
+
+    // FixedPayloadLength - 11 (header length)
+
+    [<Literal>]
+    let MaximumRelayPayloadLength = 498
+
+    // First 4 bytes of the padding for relay cell should be 0
+    [<Literal>]
+    let PaddingZeroPrefixLength = 4
+
+    [<Literal>]
+    let RelayDigestLength = 4
+
+    [<Literal>]
+    let RelayDigestOffset = 5
+
+    [<Literal>]
+    let RelayRecognizedOffset = 1
+
+    [<Literal>]
+    let RecognizedDefaultValue = 0us
+
+    //TODO: Add support for +4 tor protocol with 4 byte circuit ids
+    [<Literal>]
+    let CircuitIdLength = 2
+
+    [<Literal>]
+    let DefaultCircuitId = 0us
+
+    // Command (1 byte) + CircuitId (2 byte)
+    [<Literal>]
+    let PacketHeaderLength = 3
+
+    [<Literal>]
+    let CommandOffset = CircuitIdLength
+
+    [<Literal>]
+    let VariableLengthBodyPrefixLength = 2
+
+    [<Literal>]
+    let DefaultStreamId = 0us
```

Comment:
I think not all these constants above actually need the `[<Literal>]` attribute, right?

Reply:
^

Reply:
not addressed yet